prog/vnet/subnet_nexus.rb

```diff
@@ -58,6 +63,25 @@ def gen_reqid
     SecureRandom.random_number(100000) + 1
   end
 
+  label def update_firewall_rules
+    private_subnet.vms.each do |vm|
```

Comment:
What happens upon concurrent VM addition?

Reply:
Since the budded process works on host in the namespace of the VM, as long as the namespace setup is completed and the prep phase is done, it will be successful. If not, the budded script will run properly next time. The problem is, what would happen if a VM gets destroyed after we bud but not execute the prog just yet. I don't know that, I'll check. We might have a prog that's desperately waiting around.

Reply:
I was thinking more that it could miss a vm.

Reply:
I solved this by simply simply decrementing the semaphore before hand and implementing semaphore incr at the Nexus.assemble